drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/editor/ScenarioSimulationEditorPresenter.java

```diff
@@ -372,7 +372,14 @@ private void addRightPanelMenuItem(final FileMenuBuilder fileMenuBuilder) {
                     simpleProperties.put(modelField.getName(), modelField.getClassName());
                 }
             }
-            FactModelTree toSend = new FactModelTree(factName, packageName, simpleProperties);
+            String factPackageName;
+            String fullFactClassName = oracle.getFQCNByFactName(factName);
+            if (fullFactClassName != null &&  fullFactClassName.contains(".")) {
+                factPackageName = fullFactClassName.substring(0, fullFactClassName.lastIndexOf("."));
+            } else {
+                factPackageName = packageName;
+            }
+            FactModelTree toSend = new FactModelTree(factName, factPackageName, simpleProperties);
```

Comment:
we could shorten the code slightly, please notice you have two whitespaces between `&&` and `fullFactClassName.contains(".")`
```java
            String factPackageName = packageName;
            String fullFactClassName = oracle.getFQCNByFactName(factName);
            if (fullFactClassName != null && fullFactClassName.contains(".")) {
                factPackageName = fullFactClassName.substring(0, fullFactClassName.lastIndexOf("."));
            }
```

Also don't forget to unit tests.

Manual check pending.